core/chains/evm/txmgr/client.go

```diff
@@ -0,0 +1,154 @@
+package txmgr
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	clienttypes "github.com/smartcontractkit/chainlink/v2/common/chains/client"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/gas"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+var _ EvmTxmClient = (*evmTxmClient)(nil)
+
+type evmTxmClient struct {
+	client evmclient.Client
+}
+
+func NewEvmTxmClient(c evmclient.Client) *evmTxmClient {
+	return &evmTxmClient{client: c}
+}
+
+func (c *evmTxmClient) BatchSendTransactions(ctx context.Context, txStore EvmTxStore, attempts []EvmTxAttempt, batchSize int, lggr logger.Logger) (codes []clienttypes.SendTxReturnCode, txErrs []error, err error) {
+	// preallocate
+	codes = make([]clienttypes.SendTxReturnCode, len(attempts))
+	txErrs = make([]error, len(attempts))
+
+	reqs, batchErr := batchSendTransactions(ctx, txStore, attempts, batchSize, lggr, c.client)
+	err = errors.Join(err, batchErr) // this error does not block processing
+
+	// safety check - exits before processing
+	if len(reqs) != len(attempts) {
+		lenErr := fmt.Errorf("Returned request data length (%d) != number of tx attempts (%d)", len(reqs), len(attempts))
+		err = errors.Join(err, lenErr)
+		lggr.Criticalw("Mismatched length", "error", err)
+		return
+	}
+
+	// for each batched tx convert response to standard error code
+	for i := range reqs {
+		// convert to tx for logging purposes - exits early if error occurs
+		tx, signedErr := GetGethSignedTx(attempts[i].SignedRawTx)
```

Comment:
Is it worth doing these in parallel? Doesn't look like it would take much to convert given the current form, and this is on a hot path (plus we recently increased the default batch size).

Reply:
sounds good to me! I will address it in a later PR - https://smartcontract-it.atlassian.net/browse/BCI-1319